### PR TITLE
Forward database url from argparse to Storage

### LINKED ISSUE #2672: Pass forward parser arguments when starting storage server
Currently starting the `Storage` server through `run_ert_storage` will initialise the server without passing forward the `database-url`. The same is valid for the `run_webviz_ert` approach

https://github.com/equinor/ert/blob/a495b05edfb0798b928bbe0b37858368e23af6e7/ert_shared/main.py#L39-L40

### PR DESCRIPTION
**Pull request by lars-petter-hauge**
**Issue**
Resolves #2672 

**Approach**
Forwarded the argparse argument

## Pre review checklist

- [x] Added appropriate labels

### CODE REVIEW

File: tests/ert_tests/shared/test_main_entry.py [hunk 1]
```diff
@@ -33,3 +33,47 @@ def test_main_logging_argparse(monkeypatch, caplog):
     with caplog.at_level(logging.INFO):
         main.main()
     assert "mode='test_run'" in caplog.text
+
+
+def test_api_database_url_forwarded(monkeypatch):
```
Comment (xjules) [inline tests/ert_tests/shared/test_main_entry.py:38]: While it is good to have tests. Just wondering if it's necessary to test that the parameters are passed correctly :shrug: Just asking :)

Comment (lars-petter-hauge) [reply tests/ert_tests/shared/test_main_entry.py:38]: I know, I just felt uneasy simply modifying without any sort of tests.. Don't like the fact either than after committing only the changes, another one (e.g. myself in some x number of days...) can come and remove those changes without any tests failing..

I would think a more unified testing of the command line parsing would be more appropriate though, couldn't really find anywhere else it was tested? (except through integration tests).

I would be more than happy to modify the tests to make them more valuable though!

**Comment by lars-petter-hauge** [comment]
I have to add, there's a bit of implicit usage. And the fact that the same parsers are used from multiple angles, makes this small work a bit untidy unfortunately..

File: tests/ert_tests/shared/test_main_entry.py (hunk 1 above)
Comment (lars-petter-hauge) [reply tests/ert_tests/shared/test_main_entry.py:60]: I've thought about it a bit, and I don't think it is excessive.. In fact, I would like to add tests to extensively verify that the parser is set up correctly, and that the right path is run given input arguments.

**Comment by lars-petter-hauge** [comment]
I'm happy with this, with the exception of the failing flaky test: https://github.com/equinor/ert/issues/2635

**Comment by sondreso** [comment]
Jenkins test ert please

**Review by sondreso — APPROVED** [review]
LGTM!

**Pull request merged**